spec/requests/api/v1/admin/organizations_spec.rb

```diff
@@ -0,0 +1,102 @@
+require "rails_helper"
+
+RSpec.describe "/api/admin/organizations" do
+  let(:image) { Rails.root.join("app/assets/images/#{rand(1..40)}.png").open }
+  let!(:org_params) do
+    {
+      name: "Test Org",
+      summary: "a test org",
+      url: "https://testorg.io",
+      profile_image: image,
+      slug: "test-org",
+      tag_line: "a tagline"
+    }
+  end
+  let(:v1_headers) { { "Accept" => "application/vnd.forem.api-v1+json" } }
+
+  context "when unauthorized" do
+    let(:org_id) { create(:organization).id }
+
+    it "rejects requests without an authorization token" do
+      expect do
+        put api_admin_organization_path(org_id), params: org_params, headers: v1_headers
+      end.not_to change(User, :count)
+
+      expect(response).to have_http_status(:unauthorized)
+    end
+
+    it "rejects requests with a non-admin token" do
+      api_secret = create(:api_secret, user: create(:user))
+      headers = v1_headers.merge({ "api-key" => api_secret.secret })
+
+      expect do
+        put api_admin_organization_path(org_id), params: org_params, headers: headers
+      end.not_to change(User, :count)
+
+      expect(response).to have_http_status(:unauthorized)
+    end
+
+    it "rejects requests with a regular admin token" do
```

Comment:
I think your assumptions are probably correct! I was using the super admin on everything based on caution through uncertainty and if your impression is that it should only be deletion by super admin, it's probably coming with less uncertainty than I had when I wrote it! I'll switch to :authorize_admin in the relevant controller and set the super admin before action to :destroy only on my next push

Reply:
I'm certain about the destroy route because I just implemented it this week but regarding the :put and potentially :post I'm not 100% certain so it might also be worth double checking the assumptions with Ben/Phil if you're not really clear on it as well.